File: AadarshKumayan/question2.cpp
#include<iostream>
#include<istream>
using namespace std;
int main()
{
	long long int t[1000000],r[1000000],c[1000000],n=0,i,x=0,y,flag=0;		//t-travel time r-ready time c-cancel time
	char l='a';
	while(l!='\n')							//find out number of passengers
	{
		cin>>t[n++];
		x+=t[n-1];							//minimum time will always contain sum of all travelling time
		cin.get(l);
	}
	n++;
	for(i=0;i<n;i++)
	cin>>r[i];
	for(i=0;i<n;i++)
	cin>>c[i];
	for(i=r[0];i<=c[0];i++)						//max value between passenger 1 ready time and cancel time for which it will reach at ready time for passenger 2
	{
		if(i+t[0]<=r[1])
		y=i;
	}
	for(i=1;i<n;i++)						//calculate waiting time for all passengers
	{
	y+=t[i-1];
	if(y<r[i])
	{
		x+=r[i]-y;
		y=r[i];
	}
	else if(y>c[i])						//check if driver is late and request is not service return -1
	{
		cout<<-1<<endl;
		flag=1;
		break;
	}
	}
	if(flag==0)
	cout<<x<<endl;
	return 0;
}

File: Abhishek Dutta/question2.java
//Question 2
//Abhishek Dutta
//B.Tech CSE CCVT 4th Year, UPES Dehradun
import java.util.*;

class request{
	public int travel_time;
	public int ready_time;
	public int cancel_time;
}

public class question2 {

	static int calculate(request arr[], int N, int t_time)
	{
		int i;
		int flag = 0; //Just to check if it is possible to fulfill all requests
		for(i=1;i<N;i++)
		{
			
			//Taxi arrives early i.e before the ready time of i'th request
			if(t_time + arr[i-1].travel_time < arr[i].ready_time)
				t_time = arr[i].ready_time;
			
			//Taxi arrives on time i.e before getting cancelled 
			else if( t_time+arr[i-1].travel_time >= arr[i].ready_time && ( t_time+arr[i].travel_time )<=(arr[i].ready_time+arr[i].cancel_time))
				t_time+=arr[i].travel_time;
					
			else //Cannot fulfill any one request, then set flag to 1
			{
				flag = 1;
				break;
			}
		}
		
		if(flag==1 || t_time > 86400) //Incase total time crosses total number of seconds in a day(impossible to service)
			return -1;
		else	//Return minimum amount of time route will take
			return t_time;
		
	}
	
	public static void main(String[] args) {
		// TODO Auto-generated method stub
		int N, i, t_time;
		
		Scanner sc = new Scanner(System.in);
		N = sc.nextInt(); //
		request arr[] = new request[N];
		
		for(i=0; i< arr.length;i++) //Instantiate each element in the object array
		    arr[i] = new request();
		
		for(i=0; i<N-1; i++) //Take travel time input
		{
			arr[i].travel_time = sc.nextInt();
		}
		
		for(i=0;i<N;i++) //Take ready time and cancel time input
		{
			arr[i].ready_time = sc.nextInt();
			arr[i].cancel_time = sc.nextInt();	
		}
		t_time = arr[0].ready_time;
		System.out.println(calculate(arr,N,t_time));
	}

}
